tests/spec/tdd_spec.lua

```
@@ -0,0 +1,462 @@
+      elseif modifier == ":t:r" then return "calculator"
+      elseif modifier == ":e" then return "js"
+      end
+      return path
+    end,
+    jobstart = function(cmd, opts)
+      mock_vim._jobs = mock_vim._jobs or {}
+      local job_id = #mock_vim._jobs + 1
+      mock_vim._jobs[job_id] = {cmd = cmd, opts = opts}
+      
+      -- Simulate job completion
+      vim.schedule(function()
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'vim'

(W113)